x-pack/plugins/security_solution/public/flyout/index.tsx

```diff
@@ -90,8 +90,32 @@ const expandableFlyoutDocumentsPanels: ExpandableFlyoutProps['registeredPanels']
   },
 ];
 
+/**
+ * Flyout used for the Security Solution application
+ * We set the z-index to 997 to ensure it is always rendered behind Timeline (which has a z-index of 998)
+ */
 export const SecuritySolutionFlyout = memo(() => (
-  <ExpandableFlyout registeredPanels={expandableFlyoutDocumentsPanels} paddingSize="none" />
+  <ExpandableFlyout
+    registeredPanels={expandableFlyoutDocumentsPanels}
+    paddingSize="none"
+    customStyles={{ 'z-index': 997 }}
+  />
 ));
 
 SecuritySolutionFlyout.displayName = 'SecuritySolutionFlyout';
+
+/**
+ * Flyout used in Timeline
+ * We set the z-index to 999 to ensure it is always rendered above Timeline (which has a z-index of 998)
+ * and also behind other flyouts like Create new Case, Add Endpoint exception, Add rule exception, etc...
+ * opened from the Take action button in the footer. These latter flyouts use an overlay mask and that index is set to 1000 so we need to be below that.
+ */
```

Comment:
great idea! 👍

Reply:
I did something slightly different, instead of having those numbers as constants, I'm getting the default flyout `z-index` (`1000`) from EUI, then adding 1, 2 or 3. I did not extract this into a function because I didn't really know where to put it (the code impacted lives in different plugins. Instead I added comments everywhere to explain the decision.
@ashokaditya let me know if that is ok with you!